libopenage/config.h.in

```diff
@@ -5,6 +5,8 @@
 #ifndef OPENAGE_CONFIG_H_
 #define OPENAGE_CONFIG_H_
 
+#cmakedefine01 HAVE_THREAD_LOCAL_STORAGE
```

Comment:
No major reason. I wasn't sure what `check_cxx_source_compiles` returned. Also, `cmakedefine01` handles all the cmake's true/false cases